connectors/bitbucket-server/src/test/java/com/vmware/connectors/bitbucket/server/BitbucketServerControllerTest.java

```diff
@@ -0,0 +1,310 @@
+import static org.springframework.http.HttpMethod.POST;
+import static org.springframework.http.MediaType.*;
+import static org.springframework.test.web.client.match.MockRestRequestMatchers.method;
+import static org.springframework.test.web.client.match.MockRestRequestMatchers.requestTo;
+import static org.springframework.test.web.client.response.MockRestResponseCreators.withStatus;
+import static org.springframework.test.web.client.response.MockRestResponseCreators.withSuccess;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+public class BitbucketServerControllerTest extends ControllerTestsBase {
+
+    private static final String BITBUCKET_SERVER_AUTH_TOKEN = "bitbucket-token";
+
+    private static final String PULL_REQUEST_ID_1 = "236";
+
+    private static final String PULL_REQUEST_ID_2 = "246";
+
+    @Autowired
+    private AsyncRestTemplate rest;
+
+    private MockRestServiceServer mockBitbucketServer;
+
+    @Value("classpath:bitbucket/responses/pr_236.json")
+    private Resource pr236;
+
+    @Value("classpath:bitbucket/responses/pr_246.json")
+    private Resource pr246;
+
+    @Value("classpath:bitbucket/responses/approved.json")
+    private Resource approve;
+
+    @Value("classpath:bitbucket/responses/comments.json")
+    private Resource comments;
+
+    @Value("classpath:bitbucket/responses/declined.json")
+    private Resource declined;
+
+    @Value("classpath:bitbucket/responses/merged.json")
+    private Resource merged;
+
+    @Value("classpath:bitbucket/responses/activities_pr_236.json")
+    private Resource pr236Activities;
+
+    @Value("classpath:bitbucket/responses/activities_pr_246.json")
+    private Resource pr246Activities;
+
+    @Before
+    public void setup() throws Exception {
+        super.setup();
+
+        mockBitbucketServer = MockRestServiceServer.bindTo(rest)
+                .ignoreExpectOrder(true)
+                .build();
+    }
+
+    @Test
+    public void testProtectedResources() throws Exception {
+        testProtectedResource(POST, "/cards/requests");
+        testProtectedResource(POST, "/api/v1/UFO/app-platform-server/249/approve");
+        testProtectedResource(POST, "/api/v1/UFO/app-platform-server/249/merge");
+        testProtectedResource(POST, "/api/v1/UFO/app-platform-server/249/comments");
+        testProtectedResource(POST, "/api/v1/UFO/app-platform-server/249/decline");
+    }
+
+    @Test
+    public void testDiscovery() throws Exception {
+        testConnectorDiscovery();
+    }
+
+    @Test
+    public void testRegex() throws Exception {
+        final List<String> expectedList = ImmutableList.of(
+                // Project Name/ Repository Plug/ Pull request id.
+                "UFO/app-platform-server/244",
```

Comment:
Again, the only reason you are having to reimplement all of this code is because your expectedList is populated with just the project/repository/id.  If you instead, populate it with what the client will actually send in:

```java
expectedList = ImmutableList.of(
        "UFO/app-platform-server - Pull request #244: ",
        "UFO/app-platform-server - Pull request #245: ",
        "UFO/app-platform-server - Pull request #241: ",
        "UFO/app-platform-server - Pull request #239: ",
        "UFO/card-connectors - Pull request #9: "
);
```

then you can simply call the common test regex method.

Reply:
Changed the method as above. Now I am calling the common regex method to verify the result.